### PR TITLE
Consolidate Metadata Transform

### PR DESCRIPTION
**Pull request by norlandrhagen**
Update: Tests are passing for Zarr & Reference `ConsolidateMetadata` transform.
Update 2: Added to the PR an option for StoreToZarr to not consolidate the metadata automatically (It still defaults to consolidated_metadata).

### CODE REVIEW

**Comment by norlandrhagen** [comment]
Shoutout to @ranchodeluxe for the help with the tests!

**Comment by norlandrhagen** [comment]
Update: Tests are passing for Zarr & Reference `ConsolidateMetadata` transform.

File: pangeo_forge_recipes/writers.py [hunk 1]
```diff
@@ -67,6 +67,31 @@ def _is_first_in_merge_dim(index):
...
+
+    :param store: Input Store for Zarr or Kerchunk reference
+    :type store: MutableMapping
+    :return: Output Store
+    :rtype: MutableMapping
+    """
+
+    import zarr
+
+    # FIXME: This fails silently for Kerchunk. Does `consolidate_metadata` need a path to operate
```
Comment (ranchodeluxe) [inline pangeo_forge_recipes/writers.py:81]: remove comments

**Review by ranchodeluxe — APPROVED** [review]
LGTM!

left a reply to remove the old comments regarding the past breaking change in `consolidate_metadata`

**Comment by norlandrhagen** [comment]
It seems like regardless of if `ConsolidateMetadata` to the StoreToZarr pipeline, a `.zmetadata` is created. @cisaacstern any idea where this might be happening?

**Comment by ranchodeluxe** [comment]
> It seems like regardless of if `ConsolidateMetadata` to the StoreToZarr pipeline, a `.zmetadata` is created. @cisaacstern any idea where this might be happening?

A similar thing was happening in my PR when we are returning a `fsspec.ReferenceFileSystem` (so not exactly the same issue). [This was the change needed](https://github.com/pangeo-forge/pangeo-forge-recipes/pull/668/files#diff-8ecf7727e7fff62129c79d6f44159c0db576f7c7717f47c81a38800f5d96513dR153-R159)

**Comment by cisaacstern** [comment]
Thanks all for pushing this forward!

**Pull request merged**